src/TritonRoute/src/gc/FlexGC_main.cpp

```diff
@@ -673,8 +673,8 @@ bool FlexGCWorker::Impl::checkMetalSpacing_short_skipSameNet(
         auto yLen1 = gtl::delta(tmpRect1, gtl::VERTICAL);
         auto xLen2 = gtl::delta(tmpRect2, gtl::HORIZONTAL);
         auto yLen2 = gtl::delta(tmpRect2, gtl::VERTICAL);
-        if (xLen1 * xLen1 + yLen1 * yLen1 >= minWidth * minWidth
-            && xLen2 * xLen2 + yLen2 * yLen2 >= minWidth * minWidth) {
+        if (xLen1 * xLen1 + yLen1 * yLen1 >= minWidth * (long long) minWidth
+            && xLen2 * xLen2 + yLen2 * yLen2 >= minWidth * (long long) minWidth) {
```

Comment:
It would be better to change the variables' type to long long rather than add a cast here.  You can still get an overflow in xLen1*xLen1 or yLen1*yLen1.

Reply:
Is there a reason we don't use something like int_64t instead of long long ?

Reply:
@maliberty When I checked the type returned by gtl::delta I found it to be polygon_long_long_type as shown in the screenshot below. That's why it does not produce a warning.
![Screenshot from 2021-06-16 15-42-06](https://user-images.githubusercontent.com/56893454/122230125-a526ff00-ceb9-11eb-88f9-9c81950f8fd6.png)
@QuantamHD I can use int_64t, I chose long long to be consistent with xLen and yLen.

Reply:
I see your logic but I would go with int_64t to be consistent with how we handle area in OR.